From 2c0b5bb71aab27a3a78c1cc792cd93e1d60e3ea8 Mon Sep 17 00:00:00 2001
From: Martin Kolman <mkolman@redhat.com>
Date: Fri, 6 May 2022 12:05:57 +0200
Subject: [PATCH] Add boot options to the Web UI

There does not seem to be an easy way to do this other than using a
forked set of templates.

Changes:
- use boot options to start the Web UI & enable remote access
- skip image checksum check
- 5 second boot menu timeout
- for better debugging drop quiet boot option
- add anaconda-webui package
---
 .../config_files/x86/grub2-bios.cfg           |  6 ++---
 .../99-generic/config_files/x86/grub2-efi.cfg | 15 ++++++++-----
 .../99-generic/config_files/x86/isolinux.cfg  | 22 ++++++++++++++-----
 3 files changed, 29 insertions(+), 14 deletions(-)

diff --git a/share/templates.d/99-generic/config_files/x86/grub2-bios.cfg b/share/templates.d/99-generic/config_files/x86/grub2-bios.cfg
index cfca1bf6..cafee5d6 100644
--- a/share/templates.d/99-generic/config_files/x86/grub2-bios.cfg
+++ b/share/templates.d/99-generic/config_files/x86/grub2-bios.cfg
@@ -18,16 +18,16 @@ search --no-floppy --set=root -l '@ISOLABEL@'
 
 ### BEGIN /etc/grub.d/10_linux ###
 menuentry 'Install @PRODUCT@ @VERSION@' --class fedora --class gnu-linux --class gnu --class os {
-	linux @KERNELPATH@ @ROOT@ quiet
+	linux @KERNELPATH@ @ROOT@ inst.webui inst.webui.remote inst.resolution=1280x1024
 	initrd @INITRDPATH@
 }
 menuentry 'Test this media & install @PRODUCT@ @VERSION@' --class fedora --class gnu-linux --class gnu --class os {
-	linux @KERNELPATH@ @ROOT@ rd.live.check quiet
+	linux @KERNELPATH@ @ROOT@ rd.live.check inst.webui inst.webui.remote inst.resolution=1280x1024
 	initrd @INITRDPATH@
 }
 submenu 'Troubleshooting -->' {
 	menuentry 'Install @PRODUCT@ @VERSION@ in basic graphics mode' --class fedora --class gnu-linux --class gnu --class os {
-		linux @KERNELPATH@ @ROOT@ nomodeset quiet
+		linux @KERNELPATH@ @ROOT@ nomodeset quiet inst.resolution=1280x1024
 		initrd @INITRDPATH@
 	}
 	menuentry 'Rescue a @PRODUCT@ system' --class fedora --class gnu-linux --class gnu --class os {
diff --git a/share/templates.d/99-generic/config_files/x86/grub2-efi.cfg b/share/templates.d/99-generic/config_files/x86/grub2-efi.cfg
index 8c9adadd..e349d20a 100644
--- a/share/templates.d/99-generic/config_files/x86/grub2-efi.cfg
+++ b/share/templates.d/99-generic/config_files/x86/grub2-efi.cfg
@@ -1,4 +1,4 @@
-set default="1"
+set default="0"
 
 function load_video {
   insmod efi_gop
@@ -20,15 +20,20 @@ set timeout=60
 search --no-floppy --set=root -l '@ISOLABEL@'
 
 ### BEGIN /etc/grub.d/10_linux ###
-menuentry 'Install @PRODUCT@ @VERSION@' --class fedora --class gnu-linux --class gnu --class os {
-	linuxefi @KERNELPATH@ @ROOT@ quiet
+menuentry 'Install @PRODUCT@ @VERSION@ with Web UI' --class fedora --class gnu-linux --class gnu --class os {
+	linuxefi @KERNELPATH@ @ROOT@ inst.webui inst.webui.remote inst.resolution=1280x1024
 	initrdefi @INITRDPATH@
 }
-menuentry 'Test this media & install @PRODUCT@ @VERSION@' --class fedora --class gnu-linux --class gnu --class os {
-	linuxefi @KERNELPATH@ @ROOT@ rd.live.check quiet
+menuentry 'Test this media & install @PRODUCT@ @VERSION@ with Web UI' --class fedora --class gnu-linux --class gnu --class os {
+	linuxefi @KERNELPATH@ @ROOT@ rd.live.check quiet inst.webui inst.webui.remote inst.resolution=1280x1024
 	initrdefi @INITRDPATH@
 }
 submenu 'Troubleshooting -->' {
+	menuentry 'Install @PRODUCT@ @VERSION@ with classis GTK3 GUI' --class fedora --class gnu-linux --class gnu --class os {
+		linuxefi @KERNELPATH@ @ROOT@ quiet inst.resolution=1280x1024
+
+		initrdefi @INITRDPATH@
+	}
 	menuentry 'Install @PRODUCT@ @VERSION@ in basic graphics mode' --class fedora --class gnu-linux --class gnu --class os {
 		linuxefi @KERNELPATH@ @ROOT@ nomodeset quiet
 		initrdefi @INITRDPATH@
diff --git a/share/templates.d/99-generic/config_files/x86/isolinux.cfg b/share/templates.d/99-generic/config_files/x86/isolinux.cfg
index 216d36f2..3c516843 100644
--- a/share/templates.d/99-generic/config_files/x86/isolinux.cfg
+++ b/share/templates.d/99-generic/config_files/x86/isolinux.cfg
@@ -1,5 +1,5 @@
 default vesamenu.c32
-timeout 600
+timeout 50
 
 display boot.msg
 
@@ -58,16 +58,16 @@ menu tabmsg Press Tab for full configuration options on menu items.
 menu separator # insert an empty line
 menu separator # insert an empty line
 
-label linux
-  menu label ^Install @PRODUCT@ @VERSION@
+label linux-webui
+  menu label ^Install @PRODUCT@ @VERSION@ with Web UI
+  menu default
   kernel vmlinuz
-  append initrd=initrd.img @ROOT@ quiet
+  append initrd=initrd.img @ROOT@ inst.webui inst.webui.remote inst.resolution=1280x1024
 
 label check
   menu label Test this ^media & install @PRODUCT@ @VERSION@
-  menu default
   kernel vmlinuz
-  append initrd=initrd.img @ROOT@ rd.live.check quiet
+  append initrd=initrd.img @ROOT@ rd.live.check quiet inst.webui inst.webui.remote inst.resolution=1280x1024
 
 menu separator # insert an empty line
 
@@ -75,6 +75,16 @@ menu separator # insert an empty line
 menu begin ^Troubleshooting
   menu title Troubleshooting @PRODUCT@ @VERSION@
 
+label linux-gtk3
+  menu indent count 5
+  menu label ^Install @PRODUCT@ @VERSION@ with classic GUI
+  text help
+	Try this option if you want to use the legacy
+	@PRODUCT@ @VERSION@ GTK3 installation GUI.
+  endtext
+  kernel vmlinuz
+  append initrd=initrd.img @ROOT@ quiet inst.resolution=1280x1024
+
 label basic
   menu indent count 5
   menu label Install using ^basic graphics mode
-- 
2.41.0

